chapter_computational-performance/multiple-gpus.md

```diff
@@ -27,23 +28,23 @@ Let us start with a simple computer vision problem and a slightly archaic networ
     * We only need to synchronize after each minibatch. That said, it is highly desirable to start exchanging gradients parameters already while others are still being computed.
     * Large numbers of GPUs lead to very large minibatch sizes, thus reducing training efficiency.
 
-![Parallelization on multiple GPUs. From left to right - original problem, network partitioning, layer partitioning, data parallelism.](../img/splitting.svg)
-:label:`fig_splitting`
-
 By and large, data parallelism is the most convenient way to proceed, provided that we have access to GPUs with sufficiently large memory. See also :cite:` Li.Andersen.Park.ea.2014` for a detailed description of partitioning for distributed training. GPU memory used to be a problem in the early days of deep learning. By now this issue has been resolved for all but the most unusual cases. We focus on data parallelism in what follows.
 
 ## Data Parallelism
 
 Assume that there are $k$ GPUs on a machine. Given the model to be trained, each GPU will maintain a complete set of model parameters independently. Training proceeds as follows (see :numref:`fig_data_parallel` for details on data parallel training on two GPUs):
 
+![Calculation of minibatch stochastic gradient using data parallelism and two GPUs. ](../img/data-parallel.svg)
+:label:`fig_data_parallel`
+
```

Comment:
We should avoid inserting a figure in the middle of a paragraph. Can you move this figure just above the subtitle "Data Parallelism"?